From e6fa900e045fe0f831349dfa9612f2a3a087fb02 Mon Sep 17 00:00:00 2001
From: Sebastien Boeuf <sebastien.boeuf@intel.com>
Date: Fri, 27 Mar 2015 15:40:17 +0100
Subject: [PATCH 1/3] Add queryctrl capability to atomisp driver

This patch enhances the atomisp driver because it gives access to the "queryctrl"
ioctl defined in camera and vcm drivers.

Issue: GMINL-6562
Change-Id: I5c8a01aa2d53ed55018b2ff30f10b9d8c4525d46
Signed-off-by: Sebastien Boeuf <sebastien.boeuf@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
index 1c44af0..a8119e9 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_ioctl.c
@@ -2323,6 +2323,17 @@ static int atomisp_queryctl(struct file *file, void *fh,
 			    struct v4l2_queryctrl *qc)
 {
 	int i, ret = -EINVAL;
+	struct video_device *vdev = video_devdata(file);
+	struct atomisp_sub_device *asd = atomisp_to_video_pipe(vdev)->asd;
+	struct atomisp_device *isp = video_get_drvdata(vdev);
+
+	switch (qc->id) {
+		case V4L2_CID_FOCUS_ABSOLUTE:
+		case V4L2_CID_FOCUS_RELATIVE:
+		case V4L2_CID_FOCUS_STATUS:
+			return v4l2_subdev_call(isp->inputs[asd->input_curr].camera,
+						core, queryctrl, qc);
+	}
 
 	if (qc->id & V4L2_CTRL_FLAG_NEXT_CTRL)
 		return ret;
-- 
1.9.1

